processing/src/main/java/org/apache/druid/query/aggregation/DoubleSumAggregatorFactory.java

```diff
@@ -79,8 +81,12 @@ protected VectorValueSelector vectorSelector(VectorColumnSelectorFactory columnS
   }
 
   @Override
-  public boolean canVectorize()
+  public boolean canVectorize(ColumnCapabilitiesProvider columnCapabilitiesProvider)
   {
+    if (fieldName != null) {
+      final ColumnCapabilities originalCapabilities = columnCapabilitiesProvider.getColumnCapabilities(fieldName);
+      return expression == null && (originalCapabilities == null || originalCapabilities.getType().isNumeric());
```

Comment:
ColumnCapabilities#type is annotated as nullable. `originalCapabilities.getType().isNumeric()` could throw an NPE

Reply:
Changed to ValueType.isNumeric(originalCapabilities.getType())

Reply:
Btw I think ColumnCapabilities#type is never actually null. Segment metadata queries, which make sql work, do this to make the sql schema `capabilities.getType().name()`.
Anyway, changed to `ValueType.isNumeric(originalCapabilities.getType())` to be same and consistent with other places that use originalCapabilities.getType().